core/domain/subscription_services_test.py

```diff
@@ -162,55 +171,150 @@ def test_creating_exploration_results_in_subscription(self):
+                USER_ID), [])
+
+        COLLECTION_ID = 'col_id'
+        subscription_services.subscribe_to_collection(USER_ID, COLLECTION_ID)
+        self.assertEqual(
+            subscription_services.get_collection_ids_subscribed_to(USER_ID),
+                [COLLECTION_ID])
+
+        COLLECTION_ID_2 = 'col_id_2'
+        subscription_services.subscribe_to_collection(USER_ID, COLLECTION_ID_2)
+        self.assertEqual(
+            subscription_services.get_collection_ids_subscribed_to(USER_ID),
+            [COLLECTION_ID, COLLECTION_ID_2])
+
+    def test_creating_collection_results_in_subscription(self):
+        COLLECTION_ID = 'col_id'
+        USER_ID = 'user_id'
+
+        self.assertEqual(
+            self._get_collection_ids_subscribed_to(USER_ID), [])
+        self.save_new_default_collection(COLLECTION_ID, USER_ID)
+        self.assertEqual(
+            self._get_collection_ids_subscribed_to(USER_ID), [COLLECTION_ID])
+
+    def test_adding_new_collection_owner_or_editor_role_results_in_subscription(self):
+        COLLECTION_ID = 'col_id'
+        self.save_new_default_collection(COLLECTION_ID, self.owner_id)
+
+        self.assertEqual(
+            self._get_collection_ids_subscribed_to(self.owner_2_id), [])
+        rights_manager.assign_role_for_collection(
+            self.owner_id, COLLECTION_ID, self.owner_2_id,
+            rights_manager.ROLE_OWNER)
+        self.assertEqual(
+            self._get_collection_ids_subscribed_to(self.owner_2_id),
+            [COLLECTION_ID])
+
+        self.assertEqual(
+            self._get_collection_ids_subscribed_to(self.editor_id), [])
+        rights_manager.assign_role_for_collection(
+            self.owner_id, COLLECTION_ID, self.editor_id,
+            rights_manager.ROLE_EDITOR)
+        self.assertEqual(
+            self._get_collection_ids_subscribed_to(self.editor_id),
+            [COLLECTION_ID])
+
+    def test_adding_new_collection_viewer_role_does_not_result_in_subscription(self):
+        COLLECTION_ID = 'col_id'
+        self.save_new_default_collection(COLLECTION_ID, self.owner_id)
+
+        self.assertEqual(
+            self._get_collection_ids_subscribed_to(self.viewer_id), [])
+        rights_manager.assign_role_for_collection(
+            self.owner_id, COLLECTION_ID, self.viewer_id,
+            rights_manager.ROLE_VIEWER)
+        self.assertEqual(
+            self._get_collection_ids_subscribed_to(self.viewer_id), [])
+
+    def test_deleting_collection_does_not_delete_subscription(self):
+        COLLECTION_ID = 'col_id'
+        self.save_new_default_collection(COLLECTION_ID, self.owner_id)
+        self.assertEqual(
+            self._get_collection_ids_subscribed_to(self.owner_id),
+            [COLLECTION_ID])
+
+        collection_services.delete_collection(self.owner_id, COLLECTION_ID)
+
+        self.assertEqual(
+            self._get_collection_ids_subscribed_to(self.owner_id),
+            [COLLECTION_ID])
```

Comment:
Both are tested in a new test. I think there's good argument for both sides of whether a collection author should be subscribed to an exploration he/she includes. I'm leaning to not subscribing them for now since that's the path of least resistance; we can add it later if it seems to be a desired feature from collection authors.